Explorer/Assets/DCL/SDKComponents/AvatarShape/Systems/AvatarShapeHandlerSystem.cs

```diff
@@ -89,7 +100,108 @@ private void UpdateAvatarShape(ref PBAvatarShape pbAvatarShape, ref SDKAvatarSha
+                        emoteId,
+                        hash,
+                        bodyShape,
+                        loop: false), // looping scene emotes on SDK AvatarShapes is not supported yet
+                    newPromiseCancellationTokenComponent
+                ).Forget();
+            }
+            else
+            {
+                LoadAndTriggerSceneEmoteAsync(globalWorldEntity,
+                    new GetSceneEmoteFromRealmIntention(
+                        sceneData.SceneEntityDefinition.id!,
+                        sceneData.AssetBundleManifest,
+                        hash,
+                        loop: false, // looping scene emotes on SDK AvatarShapes is not supported yet
+                        bodyShape),
+                    newPromiseCancellationTokenComponent
+                ).Forget();
+            }
+        }
+
+        private async UniTaskVoid LoadAndTriggerSceneEmoteAsync<TIntention>(Entity globalWorldEntity, TIntention intention, SDKAvatarShapeEmotePromiseCancellationToken promiseComponent)
+            where TIntention : struct, IAssetIntention, IEquatable<TIntention>
+        {
+            CancellationToken ct = promiseComponent.Cts.Token;
+            try
+            {
+                // 1. Create the scene emote loading promise and wait for it
+                var promise = AssetPromise<EmotesResolution, TIntention>.Create(globalWorld, intention, PartitionComponent.TOP_PRIORITY);
+
+                Result<AssetPromise<EmotesResolution, TIntention>> promiseResult = await promise.ToUniTaskAsync(globalWorld, cancellationToken: ct).SuppressToResultAsync(ReportCategory.AVATAR);
+
+                if (ct.IsCancellationRequested) return;
+                if (!globalWorld.IsAlive(globalWorldEntity) || globalWorldEntity == Entity.Null) return;
+
+                // 2. Finally, add the CharacterEmoteIntent to the avatar once the emote has been loaded
+                if (promiseResult.Success)
+                {
+                    AssetPromise<EmotesResolution, TIntention> resolvedPromise = promiseResult.Value;
+
+                    if (resolvedPromise.Result is { Succeeded: true })
+                    {
+                        using var consumed = resolvedPromise.Result.Value.Asset.ConsumeEmotes();
+
+                        if (consumed.Value.Count > 0)
+                        {
+                            URN emoteUrn = consumed.Value[0]!.GetUrn();
+
+                            globalWorld.AddOrSet(globalWorldEntity,
+                                new CharacterEmoteIntent
+                                {
+                                    EmoteId = emoteUrn,
+                                    Spatial = true,
+                                    TriggerSource = TriggerSource.SCENE,
+                                });
+                        }
+                    }
+                }
+            }
+            catch (OperationCanceledException) { }
+            finally
```

Comment:
This will go away as well if you switch to the normal scheme of the promises consumption